pkgs/servers/monitoring/cockpit/default.nix

```diff
@@ -0,0 +1,152 @@
+, fetchurl
+, fetchFromGitHub
+, autoreconfHook
+, bash
+, coreutils
+, docbook_xml_dtd_43
+, docbook_xsl
+, findutils
+, gettext
+, git
+, glib
+, glibc
+, gnused
+, gnutls
+, json-glib
+, krb5
+, libssh
+, libxcrypt
+, libxslt
+, makeWrapper
+, nodejs
+, openssl
+, pam
+, pkg-config
+, polkit
+, python3Packages
+, ripgrep
+, systemd
+, udev
+, xmlto
+}:
+
+stdenv.mkDerivation rec {
+  pname = "cockpit";
+  version = "282.1";
+  enableParallelBuilding = true;
+
+  src = fetchFromGitHub {
+    owner = "cockpit-project";
+    repo = "cockpit";
+    rev = "83d2ca1295a47c4d2c5a2904a27f8087d2641657";
+    sha256 = "sha256-R3IMga5/j1tmUcXaZpzo3VhgPmsn03yI7aZUnxJouVU=";
+    leaveDotGit = true;
+    fetchSubmodules = true;
+    deepClone = true;
+  };
+
+  nativeBuildInputs = [
+    autoreconfHook
+    makeWrapper
+    docbook_xml_dtd_43
+    docbook_xsl
+    findutils
+    gettext
+    git
+    glib.dev
+    libxcrypt
+    libxslt.bin
+    nodejs
+    pam.out
+    pkg-config
+    python3Packages.python
+    python3Packages.setuptools
+    ripgrep
+    xmlto
+  ];
+
+  buildInputs = [
+    glib
+    gnutls
+    json-glib
+    krb5
+    libssh
+    polkit
+    udev
+  ];
+
+  patches = [
+    ./nerf-node-modules.patch
+    ./fix-makefiles.patch
+  ];
+
+  preConfigurePhases = [ "patchCodeImpurities" ];
+  patchCodeImpurities = ''
+    patchShebangs tools/escape-to-c
+    patchShebangs tools/make-compile-commands
+    patchShebangs tools/node-modules
+    patchShebangs tools/termschutz
+    patchShebangs tools/webpack-make
+    patchShebangs test/common/tap-cdp
+    patchShebangs test/common/pixel-tests
+    patchShebangs test/common/run-tests
+    export HOME=$(mktemp -d)
+    cp node_modules/.package-lock.json package-lock.json
+
+    substituteInPlace src/systemd_ctypes/libsystemd.py \
+      --replace libsystemd.so.0 ${systemd}/lib/libsystemd.so.0
+
+    substituteInPlace pkg/**/*.js pkg/**/*.jsx \
+      --replace '"/usr/bin/' '"' \
+      --replace '"/bin/' '"'
+
+    # for patch testing
+    # cat pkg/users/local.js
+
+    echo "m4_define(VERSION_NUMBER, [${version}])" > version.m4
+  '';
+
+  configureFlags = [
+    "--enable-prefix-only=yes"
+    "--sysconfdir=/etc"
+    "--disable-pcp" # TODO: figure out how to package its dependency
+    "--with-default-session-path=/run/wrappers/bin:/run/current-system/sw/bin"
+  ];
+
+  fixupPhase = ''
+    patchShebangs $out/libexec/cockpit-certificate-helper
+    patchShebangs $out/share/cockpit/motd/update-motd
+    patchShebangs $out/libexec/cockpit-desktop
+    PATH=${python3Packages.python.withPackages (p: with p;[ pygobject3 ])}/bin patchShebangs $out/libexec/cockpit-client
+
+    wrapProgram $out/bin/cockpit-bridge \
+      --suffix LD_LIBRARY_PATH : /run/current-system/sw/lib
+
+    wrapProgram $out/libexec/cockpit-certificate-helper \
+      --prefix PATH : ${lib.makeBinPath [ coreutils ]} \
+      --run 'cd $(mktemp -d)' \
+      --prefix PATH : ${lib.makeBinPath [ coreutils openssl ]}
```

Comment:
Is this some dark magic I don't understand?

Reply:
well, mktemp is in coreutils

Reply:
why prefix PATH is called two times?